File: src/CanvasScene.cc
#include "CanvasScene.h"

#include <cheerp/clientlib.h>

using namespace client;


CanvasScene::CanvasScene(int canvasWidth,int canvasHeight)
{
  HTMLElement * body = document.get_body();

  //add the canvas into the mix!!!!
  canvas = static_cast<HTMLCanvasElement *>(document.createElement("canvas"));
  canvas->set_width(canvasWidth);
  canvas->set_height(canvasHeight);

  context = static_cast<CanvasRenderingContext2D *>(canvas->getContext("2d"));

  //add the canvas
  body->appendChild(canvas);
}


CanvasScene::~CanvasScene()
{
  HTMLElement * body = document.get_body();

  //remove the canvas
  body->removeChild(canvas);
}

File: src/CanvasScene.js
//a super prototype for scenes that have a screen
//11/7/2015


//creates the canvas
//screenWidth is the width of the canvas
//screenHeight is the height of the canvas
var CanvasScene = function(screenWidth,screenHeight)
{
  this.canvas = document.createElement("canvas");
  this.canvas.width = screenWidth;
  this.canvas.height = screenHeight;
  document.body.appendChild(this.canvas);
  this.ctx = this.canvas.getContext("2d");
};


//deletes the canvas
CanvasScene.prototype.delete = function()
{
  document.body.removeChild(this.canvas);
};
